src/vicsetup/libvicsetup/crypto.c

```diff
@@ -0,0 +1,388 @@
+#include <string.h>
+#include <sys/time.h>
+#include <assert.h>
+#include <stdlib.h>
+
+#include <mbedtls/pkcs5.h>
+#include <mbedtls/aes.h>
+#include <mbedtls/cipher.h>
+#include <mbedtls/sha1.h>
+#include <mbedtls/sha256.h>
+#include <mbedtls/sha512.h>
+#include <mbedtls/base64.h>
+#include <mbedtls/entropy.h>
+#include <mbedtls/ctr_drbg.h>
```

Comment:
What is our story for the mbedtls dependency when this moves to userspace? Currently this is satisfied by OE's mbedtls, right?

Reply:
We are partitioning the symbol namespaces into **kernel** and **user** components. Both will contain a unique copy of OE's mbedtls.